src/practices/Cpp/CPlusPlusGitignoreCorrectlySetPractice.ts

```diff
@@ -0,0 +1,175 @@
+  private parsedGitignore: string[] = [];
+
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.CPlusPlus;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (!ctx.root.fileInspector) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const parseGitignore = (gitignoreFile: string) => {
+      return gitignoreFile
+        .toString()
+        .split(/\r?\n/)
+        .filter((content) => content.trim() !== '' && !content.startsWith('#'));
+    };
+    const content = await ctx.root.fileInspector.readFile('.gitignore');
+    const parsedGitignore = parseGitignore(content);
+    this.parsedGitignore = parsedGitignore;
+
+    // Prerequisites
+    const prerequisitesRegex = parsedGitignore.find((value: string) => /\.d$/.test(value));
+
+    // compiled object files
+    const sloRegex = parsedGitignore.find((value:string) => /\.slo$/.test(value));
+    const loRegex = parsedGitignore.find((value:string) => /\.lo$/.test(value));
+    const oRegex = parsedGitignore.find((value:string) => /\.o$/.test(value));
+    const objRegex = parsedGitignore.find((value:string) => /\.obj$/.test(value));
+
+    // precompiled headers
+    const gchHeaderRegex = parsedGitignore.find((value:string) => /\.gch$/.test(value));
+    const pchHeaderRegex = parsedGitignore.find((value:string) => /\.pch$/.test(value));
+
+    // dynamic libraries
+    const soRegex = parsedGitignore.find((value:string) => /\.so$/.test(value));
+    const dylibRegex = parsedGitignore.find((value:string) => /\.dylib$/.test(value));
+    const dllRegex = parsedGitignore.find((value:string) => /\.dll$/.test(value));
+
+    // fortran module files
+    const modRegex = parsedGitignore.find((value:string) => /\.mod$/.test(value));
+    const smodRegex = parsedGitignore.find((value:string) => /\.smod$/.test(value));
+
+    // compiled static
+    const laiRegex = parsedGitignore.find((value:string) => /\.lai$/.test(value));
+    const laRegex = parsedGitignore.find((value:string) => /\.la$/.test(value));
+    const aRegex = parsedGitignore.find((value:string) => /\.a$/.test(value));
+    const libRegex = parsedGitignore.find((value:string) => /\.lib$/.test(value));
+
+    // executables
+    const exeRegex = parsedGitignore.find((value:string) => /\.exe$/.test(value));
+    const outRegex = parsedGitignore.find((value:string) => /\.out$/.test(value));
+    const appRegex = parsedGitignore.find((value:string) => /\.app$/.test(value));
+
+    if (
+      prerequisitesRegex &&
+      sloRegex &&
+      loRegex &&
+      oRegex &&
+      objRegex &&
+      gchHeaderRegex &&
+      pchHeaderRegex &&
+      soRegex &&
+      dylibRegex &&
+      dllRegex &&
+      modRegex &&
+      smodRegex &&
+      laiRegex &&
+      laRegex &&
+      aRegex &&
+      libRegex &&
+      exeRegex &&
+      outRegex &&
+      appRegex
```

Comment:
We shouldn't check all these things. C++ works differently on every platform (operation system), so users should ignore different things on different platforms. It would be more difficult than this.